JustSaying.Models/Message.cs

```diff
@@ -19,9 +21,10 @@ protected Message()
         public string Conversation { get; set; }
         public string ReceiptHandle { get; set; }
         public string QueueUrl { get; set; }
-        public int? DelaySeconds { get; set; }
-
-        //footprint in order to avoid the same message being processed multiple times.
+        public int? DelaySeconds { get; set; }
+        public Dictionary<string, MessageAttributeValue> MessageAttributes { get; set; }

```

Comment:
Using `IDictionary` here would be nicer so the model isn't tied to a specific implementation.

Reply:
Hmmm ....

`Amazon.SimpleNotificationService.Model.PublishRequest.MessageAttributes` is tied to `Dictionary` so if we changed JustSaying.Models.Message.MessageAttributes to be an `IDictionary` we'd have to convert it to a `Dictionary` eventually.  What benefits do you see using `IDictionary` providing?

Reply:
It's just that if this is a simple contract/model that implementers are expected to derive from, using the interface gives them more flexibility, like how `IList<T>` is better than `T[]` or `List<T>`.